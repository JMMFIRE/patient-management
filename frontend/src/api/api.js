import {createApi} from '@reduxjs/toolkit/query/react'
import {fetchBaseQuery} from '@reduxjs/toolkit/dist/query/react'


// Project's base API. Will be expanded on in individual API slices.
const api = createApi({
    reducerPath: 'api',
    baseQuery: fetchBaseQuery(
        {
            baseUrl: 'http://localhost:8080'
        }),
    tagTypes: [
        'Patient', 'Task'
    ],
    endpoints: () => ({}),
})

export default api
