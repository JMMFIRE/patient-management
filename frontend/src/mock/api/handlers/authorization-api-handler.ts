import {rest} from 'msw'

const testUser = {
    userId: 1,
    email: 'massottoj@envisionpharma.com',
    password: 'test',
    firstName: 'Jacob',
    lastName: 'Massotto',
    role: 'General Doctor'
}
type LoginRequestBody = {
    email: string,
    password: string
}

export const handlers = [
    rest.post<LoginRequestBody>('http://localhost:8080/users/login', (req, res, ctx) => {
        const { email, password } = req.body

        if (email === testUser.email && password === testUser.password) {
            return res(ctx.status(200), ctx.json(testUser))
        }
        else {
            return res(ctx.status(500))
        }
    })
]
