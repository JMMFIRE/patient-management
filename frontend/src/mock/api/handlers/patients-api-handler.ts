import {rest} from 'msw'

let patientList: Patient []

/**
 * Helper method to reset the data assigned to the patientList for testing
 */
export const resetPatientList = () => {
    const initialPatientList: Patient[] = [
        {
            recordNumber: 1,
            firstName: 'Test',
            lastName: 'Patient1',
            dateOfBirth: new Date('2022-05-29T04:00:00.000+00:00'),
            sex: 'male',
            phoneNumber: '+447277713152',
            diagnosis: 'test diagnosis',
            status: 'Awaiting Surgery',
            notes: 'test notes',
            lastAppointment: new Date('2023-01-18T05:00:00.000+00:00'),
            nextAppointment: new Date ('2023-01-20T05:00:00.000+00:00'),
            doctor: {
                userId: 1,
            }
        },
        {
            recordNumber: 2,
            firstName: 'Test',
            lastName: 'Patient2',
            dateOfBirth: new Date('2022-05-29T04:00:00.000+00:00'),
            sex: 'male',
            phoneNumber: '+447277713152',
            diagnosis: 'test diagnosis',
            status: 'Recovered',
            notes: 'test notes',
            lastAppointment: new Date('2023-01-18T05:00:00.000+00:00'),
            nextAppointment: new Date ('2023-01-20T05:00:00.000+00:00'),
            doctor: {
                userId: 1,
            }
        },
        {
            recordNumber: 3,
            firstName: 'Test',
            lastName: 'Patient3',
            dateOfBirth: new Date('2022-05-29T04:00:00.000+00:00'),
            sex: 'male',
            phoneNumber: '+447277713152',
            diagnosis: 'test diagnosis',
            status: 'On Treatment',
            notes: 'test notes',
            lastAppointment: new Date('2023-01-18T05:00:00.000+00:00'),
            nextAppointment: new Date ('2023-01-20T05:00:00.000+00:00'),
            doctor: {
                userId: 1,
            }
        },
    ]

    patientList = initialPatientList
}

export const handlers = [
    rest.get('http://localhost:8080/patients/doctor/:userid', (req, res, ctx) => {
        return res(ctx.status(200), ctx.json((patientList)))
    }),
    rest.post<Patient>('http://localhost:8080/patients/patient', (req, res, ctx) => {
        const newPatient = {...req.body, recordNumber: 11}

        patientList.push(newPatient)
        return res(ctx.status(200))
    }),
    rest.delete('http://localhost:8080/patients/:patientId', (req, res, ctx) => {
        const { patientId } = req.params
        let response

        if (patientId) {
            const patientIndex = patientList.findIndex((patient) => patient.recordNumber?.toString() === patientId)
            if (patientIndex !== -1) {
                patientList.splice(patientIndex, 1)
                response = res(ctx.status(200))
            } else
                response = res(ctx.status(404))
        }
        return response
    })
]
