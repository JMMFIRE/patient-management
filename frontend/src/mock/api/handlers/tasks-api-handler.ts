import {rest} from 'msw'

let taskList: Task[]

type TaskCompletedRequest = {
    completed: boolean
}

/**
 * Helper method to reset the data assigned to the taskList for testing
 */
export const resetTaskList = () => {
    const initialTaskList: Task [] = [
        {
            taskNumber: 6383,
            title: 'Test Task',
            description: '',
            completed: false,
            createdDate: new Date('2023-02-27T05:00:00.000+00:00'),
            dueDate: new Date('2023-02-15T05:00:00.000+00:00'),
            tags: [],
            doctor: {
                userId: 1
            }
        },
        {
            taskNumber: 6384,
            title: 'Test Task 2',
            description: '',
            completed: true,
            createdDate: new Date ('2023-02-27T05:00:00.000+00:00'),
            dueDate: new Date('2023-02-15T05:00:00.000+00:00'),
            tags: ['a', 'test', 'tag'],
            doctor: {
                userId: 1
            }
        }
    ]

    taskList = initialTaskList
}

export const handlers = [
    rest.get('http://localhost:8080/tasks/doctor/:userId', (req, res, ctx) => {
        return res(ctx.status(200), ctx.json(taskList))
    }),
    rest.post<Task>('http://localhost:8080/tasks/task', (req, res, ctx) => {
        const newTask = {...req.body, taskNumber: 11}

        taskList.push(newTask)
        return res(ctx.status(200))
    }),
    rest.delete('http://localhost:8080/tasks/:taskId', (req, res, ctx) => {
        const { taskId } = req.params
        let response

        if (taskId) {
            const taskIndex = taskList.findIndex((task) => task.taskNumber?.toString() === taskId)
            if (taskIndex !== -1) {
                taskList.splice(taskIndex, 1)
                response = res(ctx.status(200))
            } else
                response = res(ctx.status(404))
        }
        return response
    }),
    rest.put<Task>('http://localhost:8080/tasks/:taskId', (req, res, ctx) => {
        const { taskId } = req.params
        const task = req.body
        let response

        if (taskId) {
            const taskIndex = taskList.findIndex((task) => task.taskNumber?.toString() === taskId)
            if (taskIndex !== -1) {
                taskList[taskIndex] = task
                response = res(ctx.status(200))
            } else
                response = res(ctx.status(404))
        }
        return response
    }),
    rest.patch<TaskCompletedRequest>('http://localhost:8080/tasks/:taskId', (req, res, ctx) => {
        const { taskId } = req.params
        const { completed } = req.body
        let response

        if (taskId) {
            const taskIndex = taskList.findIndex((task) => task.taskNumber?.toString() === taskId)
            if (taskIndex !== -1) {
                taskList[taskIndex] = {...taskList[taskIndex], completed}
                response = res(ctx.status(200))
            } else
                response = res(ctx.status(404))
        }
        return response
    })
]
