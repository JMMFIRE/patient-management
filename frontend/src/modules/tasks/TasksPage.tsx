import styles from './TasksPage.module.scss'
import TaskList from './TaskList/TaskList'
import CurrentTask from './CurrentTask/CurrentTask'
import {useState} from 'react'
import {useSelector} from 'react-redux'
import {useGetTasksQuery} from '../../features/tasks/TasksApiSlice'
import React from 'react'
import {RootState} from '../../redux/store'

const TasksPage = () => {


    const {currentUser}  = useSelector((state: RootState) => state.authentication)

    const {data: taskData, isLoading} = useGetTasksQuery(currentUser.userId)

    const [currentTask, setCurrentTask] = useState<Task | null>(null)
    const [isEditing, setIsEditing] = useState(false)

    /**
     * Sets the current task's title
     *
     * @param newTitle The new title
     */
    const setCurrentTaskTitle = (newTitle: string) => {
        setCurrentTask({...currentTask, title: newTitle} as Task)
    }

    /**
     * Sets the current task's description
     *
     * @param newDescription The new description
     */
    const setCurrentTaskDescription = (newDescription: string)  => {
        setCurrentTask({...currentTask, description: newDescription} as Task)
    }

    /**
     * Sets the current task's due date
     *
     * @param newDueDate The new due date
     */
    const setCurrentTaskDueDate = (newDueDate: Date | null)  => {
        setCurrentTask({...currentTask, dueDate: newDueDate} as Task)
    }

    /**
     * Sets the current task's tags as an array delimited by a space (" ")
     *
     * @param newTags The new tags a single String
     */
    const setCurrentTaskTags = (newTags: string)  => {
        const tags = newTags.split(' ')
        const updatedTags = tags.map(tag => tag)
        setCurrentTask({...currentTask, tags: updatedTags} as Task)
    }

    /**
     * Closes the current task view and sets current task to null
     */
    const closeCurrentTask = () => {
        setCurrentTask(null)
        setIsEditing(false)
    }

    /**
     *  Sets the current task and opens the current task view
     *
     * @param taskIndex The index of the current task to be opened
     */
    const openCurrentTask = (taskIndex: number) => {
        setCurrentTask(taskData[taskIndex])
    }

    /**
     * Opens a new blank task and sets the current task to it.
     */
    const openNewTask = () => {
        const newTaskInitialState: Task = {
            title: 'New Task',
            description: '',
            completed: false,
            createdDate: new Date(),
            dueDate: null,
            tags: [''],
            doctor: {
                userId: currentUser.userId
            }
        }

        setIsEditing(true)
        setCurrentTask(newTaskInitialState)
    }

    // We need to wait for RTK Query to complete its request before we try and load the results.
    // Here you could render a loading bar/spinner while the request is being fulfilled
    if (!isLoading) return (

        <div className={styles.page} data-testid="tasks-page">
            <div className={styles.header}>
                Tasks
            </div>
            <div className={styles.taskListView}>
                <TaskList
                    taskList={taskData}
                    onOpenCurrentTask={openCurrentTask}
                    onOpenNewTask={openNewTask}
                />
            </div>
            {currentTask ?
                <div className={styles.currentTaskView}>
                    <CurrentTask
                        task={currentTask}
                        onSetIsEditing={setIsEditing}
                        isEditing={isEditing}
                        onCloseCurrentTask={closeCurrentTask}
                        onChangeTitle={setCurrentTaskTitle}
                        onChangeDescription={setCurrentTaskDescription}
                        onChangeDueDate={setCurrentTaskDueDate}
                        onChangeTags={setCurrentTaskTags}
                    />
                </div> :
                <div/>
            }
        </div>
    )
    else
        return (
            <div>Loading</div>
        )
}

export default TasksPage
