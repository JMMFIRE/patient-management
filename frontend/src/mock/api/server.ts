import {setupServer} from 'msw/node'
import { handlers as tasksHandlers } from './handlers/tasks-api-handler'
import { handlers as authorizationHandlers } from './handlers/authorization-api-handler'
import { handlers as patientsHandlers } from './handlers/patients-api-handler'

export const server = setupServer(
    ...tasksHandlers,
    ...patientsHandlers,
    ...authorizationHandlers
)
