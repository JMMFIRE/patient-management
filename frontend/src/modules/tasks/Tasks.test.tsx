import '@testing-library/jest-dom'
import {loggedInPreloadedState, renderWithProviders} from '../../common/test/test-utils'
import TasksPage from './TasksPage'
import React from 'react'
import {fireEvent, screen, waitFor, waitForElementToBeRemoved} from '@testing-library/react'
import {server} from '../../mock/api/server'
import {resetTaskList} from '../../mock/api/handlers/tasks-api-handler'


beforeEach(() => {
    resetTaskList()
})

describe('Tasks Page', () => {

    it('Renders the tasks page, matching snapshot', async () => {
        const tasksPage = renderWithProviders(<TasksPage/>, loggedInPreloadedState)
        await waitForElementToBeRemoved(() => screen.getByText('Loading'))
        expect(tasksPage).toMatchSnapshot()

    })
    it('Renders the task page', async () => {
        renderWithProviders(<TasksPage/>, loggedInPreloadedState)
        const taskList = await screen.findByTestId('task-list')
        expect(taskList).toBeInTheDocument()
    })

    it('Populates the TaskActionbar count correctly', async () => {
        renderWithProviders(<TasksPage/>, loggedInPreloadedState)
        const actionBarTaskCount = await screen.findByTestId('task-count')
        expect(actionBarTaskCount.textContent).toEqual('2')
    })

    it('Opens a task from the task list as current task', async () =>  {
        renderWithProviders(<TasksPage/>, loggedInPreloadedState)
        const openButton = await screen.findAllByTestId('open-button')
        fireEvent.click(openButton[0])
        const saveCurrentTaskButton = await screen.findByTestId('save-current-task-button')
        expect(saveCurrentTaskButton).toBeInTheDocument()
    })

    it('Opens a new task as a current task', async () => {
        renderWithProviders(<TasksPage/>, loggedInPreloadedState)
        const newTaskButton = await screen.findByTestId('new-task-button')
        fireEvent.click(newTaskButton)
        const editTaskForm = await screen.findByTestId('edit-task-form')

        expect(editTaskForm).toBeInTheDocument()
    })

    it('Posts a new task', async () => {
        renderWithProviders(<TasksPage/>, loggedInPreloadedState)
        const newTaskButton = await screen.findByTestId('new-task-button')
        fireEvent.click(newTaskButton)

        const editTitleField = await screen.findByTestId('edit-title-field')
        const editDescriptionField = await screen.findByTestId('edit-description-field')
        const editTagsField = await screen.findByTestId('edit-tags-field')
        const saveCurrentTaskButton = await screen.findByTestId('save-current-task-button')
        const editDateField = await screen.getByPlaceholderText('Date')
        const closeButton = await screen.getByTestId('close-button')

        fireEvent.change(editTitleField, { target: { value: 'New task title'}})
        fireEvent.change(editDescriptionField, { target: { value: 'New description'}})
        fireEvent.change(editDateField, { target: { value: '04/20/2023'}})
        fireEvent.change(editTagsField, { target: { value: 'New test tags'}})
        fireEvent.click(saveCurrentTaskButton)
        fireEvent.click(closeButton)

        await waitFor(() => {
            const newTaskTitle = screen.getByText('New task title')
            expect(newTaskTitle).toBeInTheDocument()
        })
    })

    it('Deletes a task', async () => {
        renderWithProviders(<TasksPage/>, loggedInPreloadedState)

        const openButton = await screen.findAllByTestId('open-button')
        fireEvent.click(openButton[0])
        const deleteButton = await screen.findByTestId('delete-current-task-button')
        fireEvent.click(deleteButton)

        const taskToBeDeleted = screen.queryByText('Test Task')

        await waitForElementToBeRemoved(taskToBeDeleted)
    })

    it('Updates a current Task', async () => {
        renderWithProviders(<TasksPage/>, loggedInPreloadedState)

        const openButton = await screen.findAllByTestId('open-button')
        fireEvent.click(openButton[0])
        const editButton = await screen.findByTestId('edit-current-task-button')
        fireEvent.click(editButton)

        const editTitleField = await screen.findByTestId('edit-title-field')
        const editDescriptionField = await screen.findByTestId('edit-description-field')
        const editTagsField = await screen.findByTestId('edit-tags-field')
        const saveCurrentTaskButton = await screen.findByTestId('save-current-task-button')
        const editDateField = await screen.getByPlaceholderText('Date')
        const closeButton = await screen.getByTestId('close-button')

        fireEvent.change(editTitleField, { target: { value: 'Edited title'}})
        fireEvent.change(editDescriptionField, { target: { value: 'Edited description'}})
        fireEvent.change(editDateField, { target: { value: '04/21/2023'}})
        fireEvent.change(editTagsField, { target: { value: 'Edited tags'}})
        fireEvent.click(saveCurrentTaskButton)

        fireEvent.click(closeButton)

        await waitFor(() => {
            const newTaskTitle = screen.getByText('Edited title')
            expect(newTaskTitle).toBeInTheDocument()
        })
    })

    it('Sets a task completed status to true', async() => {
        renderWithProviders(<TasksPage/>, loggedInPreloadedState)

        const completedCheckboxes = await screen.findAllByTestId('completed-checkbox')
        const checkbox = completedCheckboxes[0] as HTMLInputElement

        fireEvent.click(checkbox)

        expect(checkbox.checked).toEqual(true)
    })
})
