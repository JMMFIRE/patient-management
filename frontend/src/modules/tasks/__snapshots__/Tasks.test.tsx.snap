// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasks Page Renders the tasks page, matching snapshot 1`] = `
{
  "mockedNavigate": [MockFunction],
  "renderResult": {
    "asFragment": [Function],
    "baseElement": <body>
      <div>
        <div
          class="page"
          data-testid="tasks-page"
        >
          <div
            class="header"
          >
            Tasks
          </div>
          <div
            class="taskListView"
          >
            <div
              class="taskList"
              data-testid="task-list"
            >
              <div
                class="actionBar"
              >
                <div
                  class="header"
                >
                  To be completed
                  <div
                    class="taskCount"
                    data-testid="task-count"
                  >
                    2
                  </div>
                </div>
                <div
                  class="actionMenu"
                >
                  <button
                    class="btn btn-light"
                    data-testid="new-task-button"
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4z"
                      />
                    </svg>
                  </button>
                  <button
                    class="btn btn-light"
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M1.5 1.5A.5.5 0 0 1 2 1h12a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-.128.334L10 8.692V13.5a.5.5 0 0 1-.342.474l-3 1A.5.5 0 0 1 6 14.5V8.692L1.628 3.834A.5.5 0 0 1 1.5 3.5v-2zm1 .5v1.308l4.372 4.858A.5.5 0 0 1 7 8.5v5.306l2-.666V8.5a.5.5 0 0 1 .128-.334L13.5 3.308V2h-11z"
                      />
                    </svg>
                  </button>
                  <button
                    class="btn btn-light"
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
                      />
                      <path
                        d="M5.255 5.786a.237.237 0 0 0 .241.247h.825c.138 0 .248-.113.266-.25.09-.656.54-1.134 1.342-1.134.686 0 1.314.343 1.314 1.168 0 .635-.374.927-.965 1.371-.673.489-1.206 1.06-1.168 1.987l.003.217a.25.25 0 0 0 .25.246h.811a.25.25 0 0 0 .25-.25v-.105c0-.718.273-.927 1.01-1.486.609-.463 1.244-.977 1.244-2.056 0-1.511-1.276-2.241-2.673-2.241-1.267 0-2.655.59-2.75 2.286zm1.557 5.763c0 .533.425.927 1.01.927.609 0 1.028-.394 1.028-.927 0-.552-.42-.94-1.029-.94-.584 0-1.009.388-1.009.94z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
              <div
                class="list"
              >
                <div
                  class="header"
                >
                  <div
                    class="star"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.282.95l-3.522 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
                      />
                    </svg>
                  </div>
                  New
                </div>
                <div
                  class="taskItem"
                >
                  <input
                    class="checkBox form-check-input"
                    data-testid="completed-checkbox"
                    type="checkbox"
                  />
                  <div>
                    <div
                      class="title"
                      data-testid="task-list-item-title"
                    >
                      Test Task
                      <div
                        class="dueDate"
                        data-testid="task-list-item-due-date"
                      >
                        02/15/2023
                      </div>
                    </div>
                  </div>
                  <button
                    class="options btn btn-outline-light"
                    data-testid="open-button"
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M3 9.5a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm5 0a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm5 0a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3z"
                      />
                    </svg>
                  </button>
                </div>
                <div
                  class="taskItem"
                >
                  <input
                    checked=""
                    class="checkBox form-check-input"
                    data-testid="completed-checkbox"
                    type="checkbox"
                  />
                  <div>
                    <div
                      class="title"
                      data-testid="task-list-item-title"
                    >
                      Test Task 2
                      <div
                        class="dueDate"
                        data-testid="task-list-item-due-date"
                      >
                        02/15/2023
                      </div>
                    </div>
                  </div>
                  <button
                    class="options btn btn-outline-light"
                    data-testid="open-button"
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M3 9.5a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm5 0a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm5 0a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div />
        </div>
      </div>
    </body>,
    "container": <div>
      <div
        class="page"
        data-testid="tasks-page"
      >
        <div
          class="header"
        >
          Tasks
        </div>
        <div
          class="taskListView"
        >
          <div
            class="taskList"
            data-testid="task-list"
          >
            <div
              class="actionBar"
            >
              <div
                class="header"
              >
                To be completed
                <div
                  class="taskCount"
                  data-testid="task-count"
                >
                  2
                </div>
              </div>
              <div
                class="actionMenu"
              >
                <button
                  class="btn btn-light"
                  data-testid="new-task-button"
                  type="button"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4z"
                    />
                  </svg>
                </button>
                <button
                  class="btn btn-light"
                  type="button"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M1.5 1.5A.5.5 0 0 1 2 1h12a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-.128.334L10 8.692V13.5a.5.5 0 0 1-.342.474l-3 1A.5.5 0 0 1 6 14.5V8.692L1.628 3.834A.5.5 0 0 1 1.5 3.5v-2zm1 .5v1.308l4.372 4.858A.5.5 0 0 1 7 8.5v5.306l2-.666V8.5a.5.5 0 0 1 .128-.334L13.5 3.308V2h-11z"
                    />
                  </svg>
                </button>
                <button
                  class="btn btn-light"
                  type="button"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
                    />
                    <path
                      d="M5.255 5.786a.237.237 0 0 0 .241.247h.825c.138 0 .248-.113.266-.25.09-.656.54-1.134 1.342-1.134.686 0 1.314.343 1.314 1.168 0 .635-.374.927-.965 1.371-.673.489-1.206 1.06-1.168 1.987l.003.217a.25.25 0 0 0 .25.246h.811a.25.25 0 0 0 .25-.25v-.105c0-.718.273-.927 1.01-1.486.609-.463 1.244-.977 1.244-2.056 0-1.511-1.276-2.241-2.673-2.241-1.267 0-2.655.59-2.75 2.286zm1.557 5.763c0 .533.425.927 1.01.927.609 0 1.028-.394 1.028-.927 0-.552-.42-.94-1.029-.94-.584 0-1.009.388-1.009.94z"
                    />
                  </svg>
                </button>
              </div>
            </div>
            <div
              class="list"
            >
              <div
                class="header"
              >
                <div
                  class="star"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.282.95l-3.522 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
                    />
                  </svg>
                </div>
                New
              </div>
              <div
                class="taskItem"
              >
                <input
                  class="checkBox form-check-input"
                  data-testid="completed-checkbox"
                  type="checkbox"
                />
                <div>
                  <div
                    class="title"
                    data-testid="task-list-item-title"
                  >
                    Test Task
                    <div
                      class="dueDate"
                      data-testid="task-list-item-due-date"
                    >
                      02/15/2023
                    </div>
                  </div>
                </div>
                <button
                  class="options btn btn-outline-light"
                  data-testid="open-button"
                  type="button"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M3 9.5a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm5 0a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm5 0a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3z"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="taskItem"
              >
                <input
                  checked=""
                  class="checkBox form-check-input"
                  data-testid="completed-checkbox"
                  type="checkbox"
                />
                <div>
                  <div
                    class="title"
                    data-testid="task-list-item-title"
                  >
                    Test Task 2
                    <div
                      class="dueDate"
                      data-testid="task-list-item-due-date"
                    >
                      02/15/2023
                    </div>
                  </div>
                </div>
                <button
                  class="options btn btn-outline-light"
                  data-testid="open-button"
                  type="button"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M3 9.5a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm5 0a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm5 0a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div />
      </div>
    </div>,
    "debug": [Function],
    "findAllByAltText": [Function],
    "findAllByDisplayValue": [Function],
    "findAllByLabelText": [Function],
    "findAllByPlaceholderText": [Function],
    "findAllByRole": [Function],
    "findAllByTestId": [Function],
    "findAllByText": [Function],
    "findAllByTitle": [Function],
    "findByAltText": [Function],
    "findByDisplayValue": [Function],
    "findByLabelText": [Function],
    "findByPlaceholderText": [Function],
    "findByRole": [Function],
    "findByTestId": [Function],
    "findByText": [Function],
    "findByTitle": [Function],
    "getAllByAltText": [Function],
    "getAllByDisplayValue": [Function],
    "getAllByLabelText": [Function],
    "getAllByPlaceholderText": [Function],
    "getAllByRole": [Function],
    "getAllByTestId": [Function],
    "getAllByText": [Function],
    "getAllByTitle": [Function],
    "getByAltText": [Function],
    "getByDisplayValue": [Function],
    "getByLabelText": [Function],
    "getByPlaceholderText": [Function],
    "getByRole": [Function],
    "getByTestId": [Function],
    "getByText": [Function],
    "getByTitle": [Function],
    "queryAllByAltText": [Function],
    "queryAllByDisplayValue": [Function],
    "queryAllByLabelText": [Function],
    "queryAllByPlaceholderText": [Function],
    "queryAllByRole": [Function],
    "queryAllByTestId": [Function],
    "queryAllByText": [Function],
    "queryAllByTitle": [Function],
    "queryByAltText": [Function],
    "queryByDisplayValue": [Function],
    "queryByLabelText": [Function],
    "queryByPlaceholderText": [Function],
    "queryByRole": [Function],
    "queryByTestId": [Function],
    "queryByText": [Function],
    "queryByTitle": [Function],
    "rerender": [Function],
    "unmount": [Function],
  },
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
}
`;
